moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -228,24 +227,26 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
   if (!onedof.empty())
   {
     trajectory.joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.joint_trajectory.header.stamp = stamp;
     trajectory.joint_trajectory.points.resize(waypoints_.size());
   }
 
   if (!mdof.empty())
   {
     trajectory.multi_dof_joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.multi_dof_joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.multi_dof_joint_trajectory.header.stamp = stamp;
     trajectory.multi_dof_joint_trajectory.points.resize(waypoints_.size());
   }
 
-  static const ros::Duration ZERO_DURATION(0.0);
+  static const rclcpp::Duration ZERO_DURATION(0.0);
   double total_time = 0.0;
+  rclcpp::Duration dur_total(0, 0);
   for (std::size_t i = 0; i < waypoints_.size(); ++i)
   {
     if (duration_from_previous_.size() > i)
       total_time += duration_from_previous_[i];
-
+    int seconds = static_cast<int>(total_time);
+    dur_total = rclcpp::Duration((int32_t)seconds, (int32_t)((total_time - seconds) * 1.0e+9));
```

Comment:
Maybe just a nit, but here we could also just create a `rclcpp::Duration()` of 1 second and multiply it with `total_time`.